spec.emu

```diff
@@ -70,7 +70,69 @@ contributors: Nicolò Ribaudo
         </table>
       </emu-table>
 
-      <emu-clause id="sec-module-namespace-exotic-objects-get-p-receiver" type="internal method" number="8">
+      <emu-clause id="sec-module-namespace-exotic-objects-getownproperty-p" type="internal method" number="5">
+        <h1>
+          [[GetOwnProperty]] (
+            _P_: a property key,
+          ): either a normal completion containing either a Property Descriptor or *undefined*, or a throw completion
+        </h1>
+        <dl class="header">
+          <dt>for</dt>
+          <dd>a module namespace exotic object _O_</dd>
+        </dl>
+        <emu-alg>
+          1. If <del>_P_ is a Symbol</del><ins>IsSymbolLikeNamespaceKey(_P_, _O_)</ins> is *true*, return OrdinaryGetOwnProperty(_O_, _P_).
+          1. Let _exports_ be <del>_O_.[[Exports]]</del><ins>? GetModuleExportsList(_O_)</ins>.
+          1. If _exports_ does not contain _P_, return *undefined*.
+          1. Let _value_ be ? _O_.[[Get]](_P_, _O_).
+          1. Return PropertyDescriptor { [[Value]]: _value_, [[Writable]]: *true*, [[Enumerable]]: *true*, [[Configurable]]: *false* }.
+        </emu-alg>
+      </emu-clause>
+
+      <emu-clause id="sec-module-namespace-exotic-objects-defineownproperty-p-desc" type="internal method">
+        <h1>
+          [[DefineOwnProperty]] (
+            _P_: a property key,
+            _Desc_: a Property Descriptor,
+          ): either a normal completion containing a Boolean or a throw completion
+        </h1>
+        <dl class="header">
+          <dt>for</dt>
+          <dd>a module namespace exotic object _O_</dd>
+        </dl>
+        <emu-alg>
+          1. If <del>_P_ is a Symbol</del><ins>IsSymbolLikeNamespaceKey(_P_, _O_)</ins>, return ! OrdinaryDefineOwnProperty(_O_, _P_, _Desc_).
```

Comment:
Looks like this allows this:

```js
import defer * as ns from 'mod'
Object.defineProperty(ns, 'then', { value: 1 })
ns.then // 1, not trigger evaluation
```

Is that correct?

Reply:
I don't think that would work, similarly to how `Object.defineProperty(ns, Symbol.iterator, { value: 1 })` does not currently define a `Symbol.iterator` property on traditional namespace objects.

If it works like in your code example, it's a bug in the proposal.

Reply:
Oh, you are correct, because `[[IsExtensible]]()` is false for module namespace object, so it is impossible to define a new symbol property on it.